Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,25 +1,93 @@
-function setAlarm() {}
-
-// DO NOT EDIT BELOW HERE
-
-var audio = new Audio("alarmsound.mp3");
-
-function setup() {
-  document.getElementById("set").addEventListener("click", () => {
-    setAlarm();
-  });
-
-  document.getElementById("stop").addEventListener("click", () => {
-    pauseAlarm();
-  });
-}
-
-function playAlarm() {
-  audio.play();
-}
-
-function pauseAlarm() {
-  audio.pause();
-}
-
-window.onload = setup;
+// collecting values from the user
+const headingElement = document.querySelector("#timeRemaining")
+const stopBtn = document.querySelector("#stop")
+const setBtn = document.querySelector("#set")
+
+let alarmTimer = null
+let countdownInterval = null
+
+function setAlarm() {
+  let min = document.querySelector("#alarmSet").value
+
+  if (min.trim() === "" || min <= 0) {
+    alert('Please set a valid time')
+    return;
+  }
+
+  min = Math.floor(parseInt(min))
+  let difference = min * 60 * 1000 // min to millisecond
+  let sec = min * 60 // Total seconds
+
+
+  updateCountdown(sec) // Start countdown
+
+  setBtn.disabled = true
+
+  alarmTimer = setTimeout(() => {
+    playAlarm()
+    changeBgColor()
+    setBtn.disabled = false
+  }, difference)
+}
+////////////////////////////////////////////////////////////////////////////////////////////////////////////
+function updateDisplay(seconds) {
+  let min = Math.floor(seconds / 60)
+  let remainingSeconds = seconds % 60
+  document.querySelector("#timeRemaining").innerText = `${min}:${remainingSeconds < 10 ? '0' + remainingSeconds : remainingSeconds}`

```

Comment:
```suggestion
  document.querySelector("#timeRemaining").innerText = `${min}:${remainingSeconds.padStart(2, "0"}`

```

Javascript has padStart built in for doing this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

Reply:
Noted, Thankyou.